code/modules/projectiles/projectile.dm

```diff
@@ -185,8 +224,9 @@
 				new_y--
 
 			step_towards(src, locate(new_x, new_y, z))
+			var/delay = 0

```

Comment:
Except delay is calculated two lines below, even for hitscan, so I don't really see the point of splitting.